### PR TITLE
Close transport inside handler.disconnect

### PR DESCRIPTION
**Pull request by Ma233**
Close transport inside handler.disconnect. And test it.
Try to fix the https://github.com/RingsNetwork/rings-node/issues/226

### CODE REVIEW

File: rings-core/src/message/handlers/mod.rs [hunk 1]
```diff
@@ -91,7 +92,9 @@ impl MessageHandler {
     pub async fn disconnect(&self, address: Address) -> Result<()> {
         log::info!("disconnect {:?}", address);
         self.dht.remove(address.into())?;
-        self.swarm.remove_transport(&address);
+        if let Some((_address, trans)) = self.swarm.remove_transport(&address) {
+            trans.close().await?
+        }
```
Comment (Ma233) [inline rings-core/src/message/handlers/mod.rs:97]: New logic here.

File: rings-core/src/message/handlers/connection.rs [hunk 2]
```diff
@@ -985,4 +915,79 @@ pub mod tests {
         println!("=================Finish handshake here=================");
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_finger_when_disconnect() -> Result<()> {
```
Comment (Ma233) [inline rings-core/src/message/handlers/connection.rs:920]: Test here.

File: rings-core/src/dht/finger.rs [hunk 3]
```diff
@@ -301,4 +302,90 @@ mod test {
         assert_eq!(table.len(), 0);
         assert_eq!(table.finger.len(), 3);
     }
+
+    #[test]
+    fn test_finger_table_remove_then_fill() {
```
Comment (Ma233) [inline rings-core/src/dht/finger.rs:307]: Another test for finger's removing logic inside.

**Review by Ma233 — COMMENTED** [review]
Others are just code location enhancement and some reference enhancements.

**Review by RyanKung — APPROVED** [review]

**Comment by RyanKung** [comment]
Cool

**Pull request merged**